tests/js/common/aql/aql-view-arangosearch-feature.js

```diff
@@ -523,8 +523,46 @@ function iResearchFeatureAqlTestSuite () {
         assertEqual([ "\u00F6\u00F5" ], result[0]);
         analyzers.remove(analyzerName, true);
       }
+      // no properties
+      {
+        let created = false;
+        try {
+          analyzers.save(analyzerName, "norm");
+          analyzers.remove(analyzerName, true); // cleanup (should not get there)
+          created = true;
```

Comment:
I suggest instead of setting a bool flag here to just call the `fail()` function. `fail()` will throw an exception with error text `failed`. If the code then enters the `catch` block, the assert will trigger a test failure:
```
try {
  doSomethingThatShouldFail();
  fail();
} catch (err) {
  assertSomeExpectedErrorConditionOnErr();
}
```
Note: using `fail` will make it necessary to specify it in the file's header as one of the magic global variables coming from the test environment.

Reply:
The idea was to distinguish between [no exception/wrong exception/correct exception].
But if it is common practice to use fail() for handling such cases, I will update tests accordingly.